contracts/interfaces/IAutoClaim.sol

```diff
@@ -0,0 +1,107 @@
+//SPDX-License-Identifier: CC-BY-NC-ND-4.0
+pragma solidity 0.8.7;
+
+interface IAutoClaim {
+    /**
+     * @notice Creates a notification when an auto-claim is requested
+     * @param user The user who made a request
+     * @param poolCommitter The PoolCommitter instance in which the commit was made
+     * @param updateIntervalId The update interval ID that the corresponding commitment was allocated for
+     * @param reward The reward for the auto-claim
+     */
+    event PaidClaimRequest(
+        address indexed user,
+        address indexed poolCommitter,
+        uint256 indexed updateIntervalId,
+        uint256 reward
+    );
+
+    /**
+     * @notice Creates a notification when an auto-claim request is updated. i.e. When another commit is added and reward is incremented.
+     * @param user The user whose request got updated
+     * @param poolCommitter The PoolCommitter instance in which the commits were made
+     * @param newReward The new total reward for the auto-claim
+     */
+    event PaidClaimRequestUpdate(address indexed user, address indexed poolCommitter, uint256 indexed newReward);
+
+    /**
+     * @notice Creates a notification when an auto-claim request is executed
+     * @param user The user whose request got executed
+     * @param poolCommitter The PoolCommitter instance in which the original commit was made
+     * @param reward The reward for the auto-claim
+     */
+    event PaidRequestExecution(address user, address poolCommitter, uint256 reward);
+
+    /**
+     * @notice Creates a notification when an auto-claim request is withdrawn
+     * @param user The user whose request got withdrawn
+     * @param poolCommitter The PoolCommitter instance in which the original commit was made
+     */
+    event RequestWithdrawn(address user, address poolCommitter);
+
+    struct ClaimRequest {
+        uint128 updateIntervalId; // The update interval during which a user requested a claim.
+        uint256 reward; // The amount of ETH in wei that was given by the user to pay for upkeep
+    }
+
+    function initialize(address _poolFactoryAddress) external;
+
+    /**
+     * @notice Pay for your commit to be claimed. This means that a willing participant can claim on `user`'s behalf when the current update interval ends.
+     * @dev Only callable by this contract's associated PoolCommitter instance. This prevents griefing. Consider a permissionless function, where a user can claim that somebody else wants to auto claim when they do not.
+     * @param user The user who wants to autoclaim.
```

Comment:
Isn't it bit redundant to put comments on function in both interface and implementation. Makes it so if you modify it you have to do it in two places. If we forget it will get confusing having two different comments.

Reply:
I agree that it requires two changes if the comments change, but I also think a large utility of an interface is to look at functionalities of a contract without having to also see implementations. Natspec comments assist in this.

Reply:
why not just put the comments on the interface then.

Reply:
Well because you also want to be able to have that context if you want to inspect the implementation, too.